00_Base/src/broadcaster/session.broadcaster.ts

```diff
@@ -22,34 +29,57 @@ export class SessionBroadcaster extends BaseBroadcaster {
     readonly credentialsService: CredentialsService,
   ) {
     super();
-    this.transactionRepository.transaction.on('created', (transactions) =>
-      this.broadcast(transactions),
-    );
-    this.transactionRepository.transaction.on('updated', (transactions) =>
-      this.broadcast(transactions),
-    );
+    this.transactionRepository.transaction.on('created', (transactions) => {
+      if (transactions && transactions.length > 0) {
+        this.logger.info(
+          'Attempting to broadcast created transactions',
+          transactions.map((t) => t.id),
+        );
+        this.broadcast(transactions).then();
+      }
+    });
+    this.transactionRepository.transaction.on('updated', (transactions) => {
+      if (transactions && transactions.length > 0) {
+        this.logger.info(
+          'Attempting to broadcast updated transactions',
+          transactions.map((t) => t.id),
+        );
+        this.broadcast(transactions, PutOrPatch.PATCH).then();
+      }
+    });
   }
 
-  private async broadcast(transactions: Transaction[]) {
+  private async broadcast(
+    transactions: Transaction[],
+    putOrPatch: PutOrPatch = PutOrPatch.PUT,
```

Comment:
why not just have boolean `patch`?

E.g.
```
  private async broadcast(
    transactions: Transaction[],
    patch: boolean = false,
  )
```

Reply:
adjusted